test(a3p): migrate wallet and vaults related tests from `a3p-proposals` to the `z:acceptance`

Jorge-Lopes opened the pull request:
<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

<!-- Most PRs should close a specific Issue. All PRs should at least reference one or more Issues. Edit and/or delete the following lines as appropriate (note: you don't need both `refs` and `closes` for the same one): -->

closes: https://github.com/Agoric/BytePitchPartnerEng/issues/5
closes: https://github.com/Agoric/BytePitchPartnerEng/issues/7
refs: #10049

## Description
<!-- Add a description of the changes that this PR introduces and the files that are the most critical to review. -->

This PR is part of an ongoing effort to migrate some of the selected test cases from `a3p-proposals` to the `z:acceptance` test phase. 

This particular PR new tests focus on:
- invitations and payments behavior within the smart wallet
- actions that can be executed on vaults

The source code of the migrated tests are:
- [send invitation via namesByAddress](https://github.com/Agoric/agoric-3-proposals/blob/main/proposals/75%3Aupgrade-16/provisioning.test.js#L34-L51)
- [exitOffer tool reclaims stuck payment](https://github.com/Agoric/agoric-3-proposals/blob/main/proposals/74%3Aupgrade-15/exit-reclaim.test.js#L9-L38)
- [ante handler sends fee only to vbank/reserve](https://github.com/Agoric/agoric-3-proposals/blob/main/proposals/71%3Aupgrade-14/ante-fees.test.js#L5-L74)
- Vault actions executed on [actions.sh](https://github.com/Agoric/agoric-sdk/blob/mainnet1B-rc3/packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-10/actions.sh#L105-L176)


### Security Considerations
<!-- Does this change introduce new assumptions or dependencies that, if violated, could introduce security vulnerabilities? How does this PR change the boundaries between mutually-suspicious components? What new authorities are introduced by this change, perhaps by new API calls? -->

n/a

### Scaling Considerations
<!-- Does this change require or encourage significant increase in consumption of CPU cycles, RAM, on-chain storage, message exchanges, or other scarce resources? If so, can that be prevented or mitigated? -->

n/a

### Documentation Considerations
<!-- Give our docs folks some hints about what needs to be described to downstream users.  Backwards compatibility: what happens to existing data or deployments when this code is shipped? Do we need to instruct users to do something to upgrade their saved data? If there is no upgrade path possible, how bad will that be for users? -->

One commit included in this PR fixes a small typo on `a3p-integration/proposals/README.md` 

### Testing Considerations
<!-- Every PR should of course come with tests of its own functionality. What additional tests are still needed beyond those unit tests? How does this affect CI, other test automation, or the testnet? -->

The new test files included in this PR are invoked through `test.sh`, so no changes to the current testing workflow are necessary.

However, as pointed out by @dckc in this [comment](https://github.com/bytepitch/agoric-sdk/commit/6afaba3c437c03d580ae29c1acaf7c5659f73d42#r146887047), using `waitForBlock()` can lead to unexpected behavior. To address this, we plan to update the `z:acceptance` tests to mitigate this issue.

As a solution, we propose extending the `@agoric/synthetic-chain` package to export a method like [makeRetryUntilCondition](https://github.com/Agoric/agoric-sdk/blob/f291362e4bc62ec31552cef29ec1a5f5cfbf7bd7/multichain-testing/tools/sleep.ts#L62) or a similar function. This would provide a more reliable alternative to `waitForBlock()`.

Additionally, there is already an [test-vaults.mts](https://github.com/Agoric/agoric-sdk/blob/master/a3p-integration/proposals/z%3Aacceptance/scripts/test-vaults.mts) file that test operations related to changes on auctions parameters.
To avoid confusion with `vaults.test` due to the similarity in names, I propose renaming this file to `auction.test` and expanding its scope to cover all auction-related tests.

### Upgrade Considerations
<!-- What aspects of this PR are relevant to upgrading live production systems, and how should they be addressed? -->

n/a

Chris-Hibbert commented on a3p-integration/proposals/z:acceptance/vaults.test.js:
```diff
@@ -0,0 +1,201 @@
+  const debtAfter = vaultData.debtSnapshot.debt.value;
+
+  t.true(
+    Number(debtAfter) > Number(debtBefore),
```
since this is converting to numbers anyway, it can assert that debtAfter is 1 more than debtBefore.
And the same on line 80 above.

Jorge-Lopes replied:
Thank you for bringing this to my attention.
I replaced `t.true` with `t.is` and made the difference in values explicit in the assertion, which is now also printed to the console.
Change addressed on commit [1894140](https://github.com/Agoric/agoric-sdk/pull/10123/commits/1894140c246d6935f41805d4da3b22cf592abdc2)

Chris-Hibbert commented on a3p-integration/proposals/z:acceptance/vaults.test.js:
```diff
@@ -0,0 +1,201 @@
+  const vaultData = getFormattedVaultData(vaultDataString);
+  const vaultState = vaultData.vaultState;
+
+  t.is(vaultState, 'closed', 'The vault should be in the "closed" state.');
```
Please assert that assets were returned.

Jorge-Lopes replied:
Thank you for pointing this out.

At this commit, I created an `utils.js` file that expose the `getBalance` function that is also imported by the `wallet.test.js`.
As well as other functions that are used by both tests.

Change addressed on commit [298654f](https://github.com/Agoric/agoric-sdk/pull/10123/commits/298654fda24641bf4310121c2cbf163967b54933)

Chris-Hibbert commented on a3p-integration/proposals/z:acceptance/vaults.test.js:
```diff
@@ -0,0 +1,201 @@
+  ATOM_DENOM,
+  USER1ADDR,
+  waitForBlock,
+} from '@agoric/synthetic-chain';
```
One thing I'd like to see in a long-lived acceptance test of vaults is that vaults that existed before the most recent upgrade continue to be useable.

Apparently all the vaults that were opened in earlier phases were also closed. 
@turadg, can we retroactively add opening a vault to upgrade 11?  It would be kinda nice if a3p gave us a long-lived vault to interact with.

Jorge-Lopes replied:
Thank you very much for the suggestion! 

I agree that testing vaults that existed prior to an upgrade is a relevant and important test case to include in the `z:acceptance` test phase,  to ensure proper coverage.

With this in mind, I've opened a dedicated [ticket](https://github.com/Agoric/BytePitchPartnerEng/issues/15) to ensure this scenario is included in our backlog of tests to implement.

Chris-Hibbert reviewed (changes requested):
I'd like to see a list of what test in a3p these replace.

dckc commented on a3p-integration/proposals/z:acceptance/exitOffer.js:
```diff
@@ -0,0 +1,97 @@
+  throw reason;
+};
+
+const { stringify: q } = JSON;
```
hm. how is that different from the usage below?

Chris-Hibbert replied:
it's getting `stringify` from `JSON`, and using that to encode data  for `toCapData`. I presume `JSON.stringify` does something different from the `assert.q` that sometimes prints many details from an object for logging, and other times prints few details to protect secrets or privaccy.

Jorge-Lopes replied:
I replaced the mnemonic `q` with `jq`, using as an example [net-access-fixture.js](https://github.com/Agoric/agoric-sdk/blob/master/packages/casting/test/net-access-fixture.js).
Is this a suitable solution?

Changes addressed on commit

Chris-Hibbert replied:
Hmm. you asked a question, and then (someone?) marked the question as resolved. That made it a little harder to notice the outstanding question.

I don't understand the rationale, but I don't mind the choice. It's fine, and not confusing.

dckc reviewed (commented):
IOU more info about which of these comments is critical vs. just suggestions.

dckc commented on a3p-integration/proposals/z:acceptance/wallet.test.js:
```diff
@@ -44,7 +44,7 @@ const getBalance = async (addresses, targetDenom = undefined) => {
 };
 
-test.serial(`send invitation via namesByAddress`, async t => {
+test.only(`send invitation via namesByAddress`, async t => {
```
did you mean to push this?

Jorge-Lopes replied:
No, I am working on the requested changes and by mistake end up pushing a commit with work in progress.
I already reverted that commit.

Sorry for the confusion.

Jorge-Lopes commented on a3p-integration/proposals/z:acceptance/invitation-test-submission/send-script.tjs:
```diff
@@ -0,0 +1,28 @@
+#! false node --ignore-this-line
```
Thank you for bringing this to my attention.
I did not noticed that shebang when copied this script from the proposal [75:Aupgrade-16 ](https://github.com/Agoric/agoric-3-proposals/blob/main/proposals/75%3Aupgrade-16/provisioning-test-submission/send-script.tjs)

Change addressed on commit [a492bb5](https://github.com/Agoric/agoric-sdk/pull/10123/commits/a492bb565d9fcee55c696177f0d180b81c0e4ca5)

Jorge-Lopes commented on a3p-integration/proposals/z:acceptance/test.sh:
```diff
@@ -22,3 +22,9 @@ yarn ava valueVow.test.js
+echo ACCEPTANCE TESTING wallet
+yarn ava wallet.test.js
+
+echo ACCEPTANCE TESTING vaults
```
You're absolutely right, and I apologize for the oversight. 
In my future PRs, I'll make sure to limit the scope and size. 
Thank you for your understanding

Jorge-Lopes commented on a3p-integration/proposals/z:acceptance/vaults.test.js:
```diff
@@ -0,0 +1,201 @@
+
+  const activeVaultsAfter = await agops.vaults('list', '--from', USER1ADDR);
+
+  t.true(
+    activeVaultsAfter.length === activeVaultsBefore.length,
```
I replaced `t.true` with `t.is` and made the difference in values explicit in the assertion, which is now also printed to the console.
Change addressed on commit [1894140](https://github.com/Agoric/agoric-sdk/pull/10123/commits/1894140c246d6935f41805d4da3b22cf592abdc2)

Jorge-Lopes commented on a3p-integration/proposals/z:acceptance/vaults.test.js:
```diff
@@ -0,0 +1,201 @@
+  const activeVaultsAfter = await agops.vaults('list', '--from', USER1ADDR);
+
+  t.true(
+    istBalanceBefore + 5 === istBalanceAfter,
```
Thank you for bringing this to my attention.
I replaced t.true with t.is and made the difference in values explicit in the assertion, which is now also printed to the console.
Change addressed on commit [1894140](https://github.com/Agoric/agoric-sdk/pull/10123/commits/1894140c246d6935f41805d4da3b22cf592abdc2)

Jorge-Lopes commented on a3p-integration/proposals/z:acceptance/vaults.test.js:
```diff
@@ -0,0 +1,201 @@
+    'The IST balance should increase by the minted amount',
+  );
+  t.true(
+    activeVaultsAfter.length > activeVaultsBefore.length,
```
Thank you for bringing this to my attention.
Since we are opening a single new Vault, the difference should always be 1 Vault.
Change addressed on commit [1894140](https://github.com/Agoric/agoric-sdk/pull/10123/commits/1894140c246d6935f41805d4da3b22cf592abdc2)

Jorge-Lopes commented on a3p-integration/proposals/z:acceptance/wallet.test.js:
```diff
@@ -0,0 +1,158 @@
+          '@type': moduleAcct,
+          base_account: { address },
+        },
+      } = await agd.query('auth', 'module-account', name);
```
Thank you for bringing this to my attention.
I have replace the `agd.query` and `agd.tx` calls to use the `agd` object returned by `makeAgd`
Change addressed on commit [9908477](https://github.com/Agoric/agoric-sdk/pull/10123/commits/9908477b17836a15e15f741e6a44b710f6081196)

Jorge-Lopes commented on a3p-integration/proposals/z:acceptance/vaults.test.js:
```diff
@@ -0,0 +1,201 @@
+  const { body } = JSON.parse(values[0]);
+  const cleanedBody = JSON.parse(body.slice(1));
+  return cleanedBody;
+};
```
Thank you for bringing this to my attention.
I have replaced the `queryVstorage` method with `getContractInfo`, and removed the `getFormattedVaultData` function since it is no longer required.
Change addressed on commit [641c9cd](https://github.com/Agoric/agoric-sdk/pull/10123/commits/641c9cd64443e3b3b63d0b8bfbb0a8b6279bc551)

Jorge-Lopes commented:
> I'd like to see a list of what test in a3p these replace.

I updated the PR's description to provide the list of replaced a3p tests, being them:
- [send invitation via namesByAddress](https://github.com/Agoric/agoric-3-proposals/blob/main/proposals/75%3Aupgrade-16/provisioning.test.js#L34-L51)
- [exitOffer tool reclaims stuck payment](https://github.com/Agoric/agoric-3-proposals/blob/main/proposals/74%3Aupgrade-15/exit-reclaim.test.js#L9-L38)
- [ante handler sends fee only to vbank/reserve](https://github.com/Agoric/agoric-3-proposals/blob/main/proposals/71%3Aupgrade-14/ante-fees.test.js#L5-L74)
- Vault actions executed on [actions.sh](https://github.com/Agoric/agoric-sdk/blob/mainnet1B-rc3/packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-10/actions.sh#L105-L176)

Jorge-Lopes commented on a3p-integration/proposals/z:acceptance/vaults.test.js:
```diff
@@ -0,0 +1,201 @@
+
+  const mint = '3.0';
+  const collateral = '5.0';
+  await t.throwsAsync(() => openVault(USER1ADDR, mint, collateral));
```
Thank you for pointing this out.

The error message triggered by `openVault` is considerably large (included bellow). 
Having this in mind, I thought that it would be better to assert that the error message contains the string:
`Error: Vault creation requires a minInitialDebt of {"brand":"[Alleged: IST brand]","value":"[5000000n]"}`

Do you agree with this approach?

<details>

<summary>error.message logs</summary>

Command failed with exit code 1: /usr/src/agoric-sdk/node_modules/.bin/agops perf satisfaction --from agoric1rwwley550k9mmk6uq6mm6z4udrg8kyuyvfszjk --executeOffer /tmp/agops.0Eb '--keyring-backend=test'

      {
        sharedOpts: { keyringBackend: 'test' },
        opts: {
          from: 'agoric1rwwley550k9mmk6uq6mm6z4udrg8kyuyvfszjk',
          executeOffer: '/tmp/agops.0Eb'
        }
      }
      Following :published.wallet.agoric1rwwley550k9mmk6uq6mm6z4udrg8kyuyvfszjk
      wallet update {
        currentAmount: {
          brand: Object [Alleged: SEVERED: Zoe Invitation brand] {},
          value: []
        },
        updated: 'balance'
      }
      wallet update {
        status: {
          error: 'Error: Vault creation requires a minInitialDebt of {"brand":"[Alleged: IST brand]","value":"[5000000n]"}',
          id: 'openVault-1727263883705',
          invitationSpec: {
            callPipe: [Array],
            instancePath: [Array],
            source: 'agoricContract'
          },
          proposal: { give: [Object], want: [Object] }
        },
        updated: 'offerStatus'
      }
      Error: Vault creation requires a minInitialDebt of {"brand":"[Alleged: IST brand]","value":"[5000000n]"}

      Executing  [
        '--node=http://0.0.0.0:26657',
        '--chain-id=agoriclocal',
        '--from=agoric1rwwley550k9mmk6uq6mm6z4udrg8kyuyvfszjk',
        'tx',
        'swingset',
        'wallet-action',
        '--allow-spend',
        '{"body":"#{\\"method\\":\\"executeOffer\\",\\"offer\\":{\\"id\\":\\"openVault-1727263883705\\",\\"invitationSpec\\":{\\"callPipe\\":[[\\"getCollateralManager\\",[\\"$0.Alleged: BoardRemoteATOM brand\\"]],[\\"makeVaultInvitation\\"]],\\"instancePath\\":[\\"VaultFactory\\"],\\"source\\":\\"agoricContract\\"},\\"proposal\\":{\\"give\\":{\\"Collateral\\":{\\"brand\\":\\"$0\\",\\"value\\":\\"+5000000\\"}},\\"want\\":{\\"Minted\\":{\\"brand\\":\\"$1.Alleged: BoardRemoteIST brand\\",\\"value\\":\\"+3000000\\"}}}}}","slots":["board05557","board0257"]}',
        '--keyring-backend=test',
        '--yes'
      ]
</details>

Jorge-Lopes commented on a3p-integration/proposals/z:acceptance/vaults.test.js:
```diff
@@ -0,0 +1,201 @@
+  const mint = '5.0';
+  const collateral = '10.0';
+  await openVault(USER1ADDR, mint, collateral);
+  await waitForBlock();
```
Thank you for pointing this out.

The `waitForBlock` is unnecessary here when following `openVault`
Change addressed on commit [78de723](https://github.com/Agoric/agoric-sdk/pull/10123/commits/78de7233bc3cfd22cec84d63747fbb810b11344c)

Jorge-Lopes commented on a3p-integration/proposals/z:acceptance/wallet.test.js:
```diff
@@ -0,0 +1,158 @@
+  t.log('uist balance before:', before);
+
+  await $`node ./exitOffer.js --id ${offerId} --from ${from} `;
+  await waitForBlock(2);
```
If I understand the request correctly, the goal is to retrieve the `offerStatus` of the offer submitted by the `exitOffer` script, in order to verify whether the offer has completed or failed.

To implement this, I assume we could use a polling mechanism similar to the [pollOffer](https://github.com/Agoric/agoric-sdk/blob/master/packages/agoric-cli/src/lib/wallet.js#L256-L286) method, which is defined in `packages/agoric-cli/src/lib/wallet.js`.

However, I noticed that in `z:acceptance/test-lib/wallet.js`, there’s a [comment](https://github.com/Agoric/agoric-sdk/blob/master/a3p-integration/proposals/s%3Astake-bld/test-lib/wallet.js#L1-L4) stating that the dependencies for these features were removed due to import issues:

```
/**
 * @file copied from packages/agoric-cli,
 * removing polling and coalescing features whose dependencies cause import problems here
 */
 ```
 
@dckc @turadg Could you please confirm whether this limitation still applies? If so, is there an alternative approach you'd recommend to poll and verify the offerStatus without reintroducing those import issues?

Jorge-Lopes replied:
After discussing this during the Office Hours session, we've decided that the replacement of waitForBlock call with a more appropriate function would be a suitable solution for this vulnerability.

This task is already scheduled and you can track its progress via the following [issue](https://github.com/Agoric/BytePitchPartnerEng/issues/10).

Given this, I'm unsure if the correct procedure is to merge the PR as it is and address the changes in a subsequent update, or keep the PR open until those changes are incorporated.

Note: the same applies for the [requested changes](https://github.com/Agoric/agoric-sdk/pull/10123#discussion_r1772319163) above.

dckc replied:
> I'm unsure if the correct procedure is to merge the PR as it is and address the changes in a subsequent update

Given that `waitForBlock` work is tracked in other issues, let's leave it out of scope for this PR; that is: use the former procedure: subsequent update.

Jorge-Lopes commented on a3p-integration/proposals/z:acceptance/exitOffer.js:
```diff
@@ -0,0 +1,97 @@
+  (offerId && from) || badUsage();
+
+  env.PATH = `${bin}:${env.PATH}`;
+  const action = { method: 'tryExitOffer', offerId };
+  const out = await doAction(action, from);
```
See the [comment](https://github.com/Agoric/agoric-sdk/pull/10123#discussion_r1775752882) bellow

Chris-Hibbert commented on a3p-integration/proposals/z:acceptance/vaults.test.js:
```diff
@@ -0,0 +1,250 @@
+
+test.serial('open new vault', async t => {
+  const istBalanceBefore = await getISTBalance(USER1ADDR);
+  const activeVaultsBefore = await agops.vaults('list', '--from', USER1ADDR);
```
please make a helper. 
```
const agopsVaults = addr => agops.vaults('list', '--from', addr);
```

Jorge-Lopes replied:
Changes addressed on commit [b98e148](https://github.com/Agoric/agoric-sdk/pull/10123/commits/b98e148bb46da4d7457f21cf0f0477652a3eda15)

Chris-Hibbert commented on a3p-integration/proposals/z:acceptance/vaults.test.js:
```diff
@@ -0,0 +1,250 @@
+  t.log('vault collateral after:', collateralAfter);
+
+  t.is(
+    Number(collateralBefore),
+    Number(collateralAfter) + 1000000,
```
BigInts compare just fine. Please don't convert to `Number` unnecessarily.
```suggestion
    collateralBefore,
    collateralAfter + 1_000_000n,
```

(and similar tests later.)

dckc commented on a3p-integration/proposals/z:acceptance/invitation-test-submission/send-script.tjs:
```diff
@@ -0,0 +1,27 @@
+/// <reference types="@agoric/vats/src/core/core-eval-env"/>
+/// <reference types="@agoric/vats/src/core/types-ambient"/>
+
+/**
+ * Send a payment by looking up deposit facet via namesByAddress.
+ *
+ * see ./post.test.js
+ *
+ * @param {BootstrapPowers} powers
+ */
+const sendIt = async powers => {
```
hm. what's misleading about the comment? The purpose of the function is to verify that namesByAddress is working.

The fact that the payment is an invitation to the reserve is incidental. It's just an easy payment to construct.

Chris-Hibbert replied:
> The purpose of the function is to verify that namesByAddress is working.

> The fact that the payment is an invitation to the reserve is incidental. It's just an easy payment to construct.

In that case a few comments would clarify. 
 * mention that this file is used as a template, and that **ADDRESS** is edited before use
 * mention that the use of the reserve is incidental since the purpose is testing `namesByAddress`.

Jorge-Lopes replied:
Changes addressed on commit [e0fa6c3](https://github.com/Agoric/agoric-sdk/pull/10123/commits/e0fa6c35b3898f6bc5dd30659754545953cea371)

Jorge-Lopes commented on a3p-integration/proposals/z:acceptance/vaults.test.js:
```diff
@@ -0,0 +1,250 @@
+      'Error: Vault creation requires a minInitialDebt of {"brand":"[Alleged: IST brand]","value":"[5000000n]"}',
+    ),
+    'Error message does not contain the expected text',
+  );
```
I missed the earlier suggestion regarding the use of regular expressions made by @dckc on this [comment](https://github.com/Agoric/agoric-sdk/pull/10123#discussion_r1772309612) for the error message. I wasn't aware that a regular expression could be provided as the expectation in t.throwsAsync(), which is why I overlooked it initially.

I also apologize for prematurely resolving the conversation. I understand that it made it harder to track the changes made and open questions. Moving forward, I will make sure to leave conversations open and wait for the original commenter to resolve them.

Thank you for your patience and guidance.

Changes addressed on commit [2f7d52d](https://github.com/Agoric/agoric-sdk/pull/10123/commits/2f7d52d4d4d293c1287bcdc4a36f86d49ffc46e4)

Jorge-Lopes commented on a3p-integration/proposals/z:acceptance/test-lib/utils.js:
```diff
@@ -0,0 +1,46 @@
+
+      if (targetDenom) {
+        const balance = balances.find(({ denom }) => denom === targetDenom);
+        return Number(balance.amount);
```
Changes addressed on commit [2afeb25](https://github.com/Agoric/agoric-sdk/pull/10123/commits/2afeb253b4fc99adcd5ca68b378ef31f37f652a0)

Jorge-Lopes commented on a3p-integration/proposals/z:acceptance/test-lib/utils.js:
```diff
@@ -0,0 +1,46 @@
+ * @param {string[]} addresses
+ * @param {string} [targetDenom]
+ */
+export const getBalance = async (addresses, targetDenom = undefined) => {
```
Changes addressed on commit [2afeb25](https://github.com/Agoric/agoric-sdk/pull/10123/commits/2afeb253b4fc99adcd5ca68b378ef31f37f652a0)

Chris-Hibbert commented on a3p-integration/proposals/z:acceptance/invitation-test-submission/send-script.tjs:
```diff
@@ -0,0 +1,27 @@
+/**
+ * Send a payment by looking up deposit facet via namesByAddress.
+ *
+ * see ./post.test.js
```
Did this  get renamed?  I see a reference in `wallet.test.js`.

Chris-Hibbert commented on a3p-integration/proposals/z:acceptance/vaults.test.js:
```diff
@@ -21,22 +21,21 @@ test.serial('attempt to open vaults under the minimum amount', async t => {
+      message:
+        /Vault creation requires a minInitialDebt of {"brand":"\[Alleged: IST brand\]","value":"\[5000000n\]"}/,
+    },
+    'Error triggered by openVault does not contain the expected message',
```
Your choice whether to leave it or not, but I don't think this adds much to the default error message which would also indicate that the resulting message didn't match.

Jorge-Lopes replied:
Changes addressed on commit [4216ee1](https://github.com/Agoric/agoric-sdk/pull/10123/commits/4216ee1d9c560d7581f600009a8a9de321aa43d0)

Chris-Hibbert commented on a3p-integration/proposals/z:acceptance/vaults.test.js:
```diff
@@ -0,0 +1,248 @@
+
+  t.is(
+    collateralBefore,
+    collateralAfter - 1000000n,
```
using underscores to separate groups of digits is optional, but I find counting strings of zeros by eye to be error prone.

Javascript ignores underscores in numbers, so we often mimic the American English practice of separating triples with commas: 1,000,000.

```suggestion
    collateralAfter - 1_000_000n,
```

Jorge-Lopes replied:
Changes addressed on commit [4216ee1](https://github.com/Agoric/agoric-sdk/pull/10123/commits/4216ee1d9c560d7581f600009a8a9de321aa43d0)

Jorge-Lopes commented:
I noticed that the original pull request included merge commits due to fetching the state of the upstream repository.
To align with the guidelines to to keep the commit history clean, I have rebased my branch against the upstream master branch and removed the unnecessary merge commits. 
This process resulted in the PR being closed and reopened.

I am sorry for any inconvenience and please let me know if anything else is needed.
Thank you.

Chris-Hibbert commented:
We don't usually require a re-review after rebasing for merge. Did anything interesting change? If not, my previous approval holds.

Chris-Hibbert commented:
@LuqiPan pointed out that the merge process may require more assistance from us. Sorry, I didn't realize that. We (Luqi) are looking into that.

Jorge-Lopes commented:
> @LuqiPan pointed out that the merge process may require more assistance from us. Sorry, I didn't realize that. We (Luqi) are looking into that.

Thank you for the feedback. 
If there's anything else we can do to help with the merge process, just let us know.
